server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/remote/delivery/RemoteDeliveryConfiguration.java

```diff
@@ -281,6 +288,14 @@ public Collection<String> getGatewayServer() {
         return gatewayServer;
     }
 
+    public synchronized Collection<String> getGatewayServerRotateList() {
+        if (!gatewayServer.isEmpty() && gatewayServer.size() > 0) {
+            Collections.rotate(gatewayServerRotateList,1);
+            return List.copyOf(gatewayServerRotateList);
+        }
+        return List.copyOf(gatewayServerRotateList);
+    }
```

Comment:
Here we have a design problem.

 -> a Configuration is meant as "immutable" and represent the content of the conf file.
 
 It should be absent of application logic and shouldn't  hold state.
 
 The use of `synchronized` looks like an uneeded bottleneck to me.
 
 I would suggest just copying the collection of gateways and shuffling it for each outgoing email
  -> no state
  -> no synchronization.
  
  Thoughts?

Reply:
My first solution was with shuffle as well but i preferred a rotation. i switched it back to a shuffle model